crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -2735,10 +2735,22 @@ impl Dependencies {
             let mut new_forks: Vec<Fork> = vec![];
             if let Some(markers) = fork_groups.remaining_universe() {
                 trace!("Adding split to cover possibly incomplete markers: {markers}");
-                new_forks.push(Fork {
-                    dependencies: vec![],
-                    markers,
-                });
+                let mut new_forks_for_remaining_universe = forks.clone();
```

Comment:
Shouldn't this be a single fork that we build by using the original deps and removing all that are disjoint with this fork's markers?

Reply:
I don't think so. That was how I had originally done it (except without adding original deps). Crucially, this new "remaining universe" partitioning is added in the same way as other partitionings are. That is, it needs to be considered in the context of any extant forks. Basically, when a fork is created, it starts by copying all extant forks at the time the partitioning was determined, adding dependencies corresponding to the fork along with the markers.

Let me see if I can come up with an example using packse.

Reply:
OK, I _think_ I came up with an example here: https://github.com/astral-sh/packse/pull/204

For that scenario, I believe your suggestion would lead to the "remaining universe" fork created by `b` in the dependencies of `a` having a marker expression of just `os_name != 'linux' and os_name != 'darwin'`. But it actually should be combined with the marker expressions of every parent fork.

Reply:
The scenario doesn't seem to use that code path, it looks like with it we're only doing one iteration with the parent universal fork:

```rust
if let Some(markers) = fork_groups.remaining_universe() {
    trace!("Adding split to cover possibly incomplete markers: {markers}");
    println!(
        "forks: {} {}",
        forks.len(),
        forks
            .iter()
            .map(|fork| format!("({})", fork.markers))
            .join(" || ")
    );
    let mut new_forks_for_remaining_universe = forks.clone();
    for fork in &mut new_forks_for_remaining_universe {
        fork.markers.and(markers.clone());
        fork.dependencies.retain(|dep| {
            let Some(dep_marker) = dep.package.marker() else {
                return true;
            };
            // After we constrain the markers on an existing
            // fork, we should ensure that any existing
            // dependencies that are no longer possible in this
            // fork are removed. This mirrors the check we do in
            // `add_nonfork_package`.
            !crate::marker::is_disjoint(&fork.markers, dep_marker)
        });
    }
    println!(
        "new_forks_for_remaining_universe: {} {}",
        new_forks_for_remaining_universe.len(),
        new_forks_for_remaining_universe
            .iter()
            .map(|fork| format!("({})", fork.markers))
            .join(" || ")
    );
    new_forks.extend(new_forks_for_remaining_universe);
}
```

```console
$ rm -f uv.lock && cargo run lock -v 2> a.txt
forks: 1 ()
new_forks_for_remaining_universe: 1 (sys_platform != 'illumos' and sys_platform != 'windows')
forks: 1 ()
new_forks_for_remaining_universe: 1 (os_name != 'darwin' and os_name != 'linux')
```

Reply:
We will merge this as-is and follow-up on it later, overlapping markers are more important to fix.

Reply:
You're 100% correct. It doesn't! I even had my own debug prints and was not careful in looking at what they were telling me. I spent some time trying to up with a better example, but failed. I filed https://github.com/astral-sh/uv/issues/5482 to track investigation into this possible issue.